UndertaleModTool/MainWindow.xaml.cs

```diff
@@ -2526,10 +2528,21 @@ public string ProcessException(in Exception exc, in string scriptText)
                 return $"An error occurred while processing the exception text.\nError message - \"{e.Message}\"\nThe unprocessed text is below.\n\n" + excString;
             }
 
-            if (excLineNums.Count > 0) //if line number(s) is found
+            if (loadedScriptLineNums.Count > 0) //if line number(s) is found
             {
-                string[] scriptLines = scriptText.Split('\n');
-                string excLines = string.Join('\n', excLineNums.Select(n => $"Line {n}: {scriptLines[n].TrimStart(new char[] { '\t', ' ' })}"));
+                // read the code for the files to know what the code line associated with the stack trace is
+                Dictionary<string, List<string>> scriptsCode = new();
+                foreach (Tuple<string, int> scriptLineNum in loadedScriptLineNums)
+                {
+                    string scriptPath = scriptLineNum.Item1;
+                    if (!scriptsCode.ContainsKey(scriptPath))
+                    {
+                        string scriptCode = File.ReadAllText(scriptPath);
+                        scriptsCode.Add(scriptPath, scriptCode.Split('\n').ToList());
+                    }
+                }
+
+                string excLines = string.Join('\n', loadedScriptLineNums.Select(pair => $"Line {pair.Item2} in script {pair.Item1}: {scriptsCode[pair.Item1][pair.Item2 - 1]}")); // - 1 because line numbers start from 1
```

Comment:
```suggestion
                string excLines = string.Join('\n', loadedScriptLineNums.Select(pair => 
                {
                    string scriptName = pair.Item1.Split(Path.DirectorySeparatorChar)[^1];
                    return $"Line {pair.Item2} in script {scriptName}: {scriptsCode[pair.Item1][pair.Item2 - 1]}"; // - 1 because line numbers start from 1
                }));
```